src/progress.rs

```diff
@@ -0,0 +1,82 @@
+use colored::Colorize;
+use git2::{Error, Progress};
+use indicatif::{HumanBytes, MultiProgress, ProgressBar, ProgressStyle};
+
+use crate::git::GitAction;
+
+pub struct ProgressTracker(MultiProgress);
+
+impl ProgressTracker {
+    pub fn new() -> Self {
+        let bars = MultiProgress::new();
+        bars.set_move_cursor(true);
+        Self(bars)
+    }
+
+    pub fn new_bar(&self, remote_url: &str) -> ProgressReporter {
+        let prog_bar = self
+            .0
+            .add(ProgressBar::new_spinner())
+            .with_style(ProgressStyle::default_bar().template(STYLE_PRELOAD));
+
+        prog_bar.set_prefix(&remote_url.blue());
+        prog_bar.set_message(&"Waiting for process to begin".cyan());
+        prog_bar.set_draw_delta(100);
+
+        ProgressReporter(prog_bar)
+    }
+
+    pub fn start_task(&self, mut action: impl GitAction + Send + 'static) {
+        let progress_bar = self.new_bar(&action.get_name());
+        rayon::spawn(move || action.do_git_action(progress_bar));
+    }
+
+    pub fn join(self) -> Result<(), std::io::Error> {
+        self.0.join()
+    }
+
+    // internalAutoTick ?
+}
+
+pub struct ProgressReporter(ProgressBar);
+
+const STYLE_PRELOAD: &str = "{prefix:>40!.blue} {spinner} {wide_msg:.cyan}";
+const STYLE_LOAD: &str = "{prefix:>40!.blue} {msg} {wide_bar} {percent:>3}% {eta}";
+const STYLE_DONE: &str = "{prefix:>40!.blue} {wide_msg} {elapsed_precise}";
+const STYLE_ERROR: &str = "{prefix:>40!.blue} {wide_msg:.red}";
+
```

Comment:
Can we make the alignment either completely left aligned or justified?

Reply:
Left align : (use ^ for centering)
```suggestion
const STYLE_LOAD: &str = "{prefix:<40!.blue} {msg} {wide_bar} {percent:>3}% {eta}";
const STYLE_DONE: &str = "{prefix:<40!.blue} {wide_msg} {elapsed_precise}";
const STYLE_ERROR: &str = "{prefix:<40!.blue} {wide_msg:.red}"; 
```

Reply:
```suggestion
const STYLE_PRELOAD: &str = "{prefix:.blue} {spinner} {wide_msg:.cyan}";
const STYLE_LOAD: &str = "{prefix:.blue} {msg} {wide_bar} {percent:>3}% {eta}";
const STYLE_DONE: &str = "{prefix:.blue} {wide_msg} {elapsed_precise}";
const STYLE_ERROR: &str = "{prefix:.blue} {wide_msg:.red}";
```
I would suggest not giving a size to the prefix. The paths can be of any length and will be truncated if it doesnt fit right now. Let it always pruint the entire apth and then the message.

Reply:
Can you do this change as part of ur fork as well? Or should I make the change once I merge?